source/extensions/filters/network/zookeeper_proxy/zookeeper_utils.cc

```diff
@@ -0,0 +1,53 @@
+#include "extensions/filters/network/zookeeper_proxy/zookeeper_utils.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ZooKeeperProxy {
+
+bool BufferHelper::peekInt32(Buffer::Instance& buffer, uint64_t& offset, int32_t& val) {
+  try {
+    val = buffer.peekBEInt<int32_t>(offset);
+    offset += sizeof(uint32_t);
+    return true;
+  } catch (EnvoyException& e) {
+    return false;
```

Comment:
Is this the right way to handle this exception? From looking at the code, it seems to only happen if you try to read off the end up of the buffer. Seems like this would indicate a protocol error, so did you consider at least recording an error here?

Reply:
Yeah let me clean this up to consistently handle bound checks, possible protocol errors and things that we can just ignore.